src/check_datapackage/read_json.py

```diff
@@ -28,11 +28,13 @@ def read_json(path: Path) -> dict[str, Any]:
             pos=error.pos,
         ) from None  # To hide the original traceback
 
-    if not isinstance(descriptor, dict):
+    if not isinstance(properties, dict):
         raise TypeError(
             f"The file {path} should parse into a Python dictionary (`dict`) "
-            f"but it converts to the type `{type(descriptor)}`. Is the file "
+            f"but it converts to the type `{type(properties)}`. Is the file "
             "missing a curly bracket `{` at the beginning or `}` at the end?"
-        )
+        ) from None  # To hide the original traceback
 
-    return descriptor
+    # Can't determine type at end from the above input and processing,
+    # so ignore the pyright/pylance warning.
+    return properties  # pyright: ignore[reportUnknownVariableType]
```

Comment:
Ohhh, are you running Pylance/Pyright in strict mode? I was running it in standard mode, so I didn't see this error. Should I be running it in strict?

I think it's safe to cast at this point and that will make Pylance happy:
```suggestion
    return cast(dict[str, Any], properties)
```